### ISSUE TITLE
ansible.cfg does not recognize $HOME properly (edit: in SSH args)

### ISSUE THREAD

**Issue by DrStrangepork** [ansible/ansible#9608]
With these values set in my $HOME/.ansible.cfg:

> private_key_file = $HOME/.ssh/keypair.pem

The output for the command `ansible-playbook -vvvv --check -i hosts site.yml` include:

> <host> EXEC ['ssh', '-C', '-tt', '-vvv', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/rkasten/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'Port=22', '-o', '**IdentityFile="$HOME/.ssh/keypair.pem"**', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'User=keypair', '-o', 'ConnectTimeout=10', 'host', "/bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1416843520.79-15764353703490 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1416843520.79-15764353703490 && echo $HOME/.ansible/tmp/ansible-tmp-1416843520.79-15764353703490'"]

And in the ssh debug section, there is this:

> debug1: Trying private key: $HOME/.ssh/keypair.pem
> no such identity: $HOME/.ssh/keypair.pem: No such file or directory

However, the command `ansible-playbook -vvvv --check --private-key $HOME/.ssh/keypair.pem -i hosts site.yml` works fine with this output:

> <host> EXEC ['ssh', '-C', '-tt', '-vvv', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/rkasten/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'Port=22', '-o', '**IdentityFile="/home/rkasten/.ssh/keypair.pem"**', '-o', 'KbdInteractiveAuthentication=no', '-o', 'PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey', '-o', 'PasswordAuthentication=no', '-o', 'User=keypair', '-o', 'ConnectTimeout=10', 'host', "/bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1416843941.97-204104573912616 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1416843941.97-204104573912616 && echo $HOME/.ansible/tmp/ansible-tmp-1416843941.97-204104573912616'"]

Is this by design?

**Comment by mpdehaan** [comment]
It's not.

Noted - This is a feature request / buglet - desire to expand shell variables inside SSH args.

Thanks!

**Comment by madduck** [comment]
I would like to note that the default config specifies

```
remote_tmp     = $HOME/.ansible/tmp
```

but when I try to use `$HOME` e.g. for `hostfile`, it does not get expanded (and neither does `~`):

```
hostfile       = $HOME/sa/ansible/inventory
```

**Comment by emdete** [comment]
for me ~ gets expanded and even worse: it will be expanded for remote_tmp as well which is a really bad idea as local home is probably different from remote home.
[using ansible 1.7.2+dfsg-2 on debian]

**Comment by jctanner** [comment]
This was fixed in 1.8
